go/inst/instance_dao.go

```diff
@@ -380,8 +380,14 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
 
 	latency.Start("instance")
 	db, err := db.OpenDiscovery(instanceKey.Hostname, instanceKey.Port)
-	latency.Stop("instance")
-	if err != nil {
+	if err == nil {
```

Comment:
Not a strong suggestion, but consider the following flow which at least for me seems to more straightforward and clean:
```	
        db, err := db.OpenDiscovery(instanceKey.Hostname, instanceKey.Port)
	if err != nil {
		latency.Stop("instance")
		goto Cleanup
	}

	err = db.Ping()
	latency.Stop("instance")
	if err != nil {
		goto Cleanup
	}
```